src/redux/services/unsavedResource.ts

```diff
@@ -35,16 +35,30 @@ export function createUnsavedResource(
   let newResourceContent: any;
 
   if (jsonTemplate) {
-    newResourceContent = {
-      ...jsonTemplate,
-      apiVersion: input.apiVersion,
-      kind: input.kind,
-      metadata: {
-        ...(jsonTemplate.metadata || {}),
-        name: input.name,
-        namespace: input.namespace,
-      },
-    };
+    if (jsonTemplate.kind && jsonTemplate.apiVersion) {
```

Comment:
We could use directly this here so we don't duplicate the object:

```
newResourceContent = {
        ...jsonTemplate,
        apiVersion: input.apiVersion,
        kind: input.kind,
        metadata: {
          ...(jsonTemplate.metadata || {}),
          name: input.name,
          namespace: input.namespace,
        },
        ...(jsonTemplate.kind && jsonTemplate.apiVersion && {...jsonTemplate}),
      };
```

Reply:
the problem I had with putting jsonTemplate first was that the apiVersion/kind/metadata properties were added to the end of the document - which was weird in the editor